mllib/src/test/scala/org/apache/spark/ml/regression/LinearRegressionSuite.scala

```diff
@@ -93,525 +96,641 @@ class LinearRegressionSuite extends SparkFunSuite with MLlibTestSparkContext {
+          intercept[IllegalArgumentException] {
+            trainer1a.fit(dataset)
+            trainer1a.fit(weightedData)
+            trainer1b.fit(weightedData)
+          }
+          break
+        } else {
+          model1a0 = trainer1a.fit(data)
+          model1a1 = trainer1a.fit(weightedData)
+          model1b = trainer1b.fit(weightedData)
+        }
+
+        assert(model1a0.weights !~= model1a1.weights absTol 1E-3)
+        assert(model1a0.intercept !~= model1a1.intercept absTol 1E-3)
+        assert(model1a0.weights ~== model1b.weights absTol 1E-3)
+        assert(model1a0.intercept ~== model1b.intercept absTol 1E-3)
+
+        val trainer2a = (new LinearRegression).setFitIntercept(true)
+          .setElasticNetParam(0.38).setRegParam(0.21).setStandardization(false).setSolver(solver)
+        val trainer2b = (new LinearRegression).setFitIntercept(true).setWeightCol("weight")
+          .setElasticNetParam(0.38).setRegParam(0.21).setStandardization(false).setSolver(solver)
+        val model2a0 = trainer2a.fit(data)
+        val model2a1 = trainer2a.fit(weightedData)
+        val model2b = trainer2b.fit(weightedData)
+        assert(model2a0.weights !~= model2a1.weights absTol 1E-3)
+        assert(model2a0.intercept !~= model2a1.intercept absTol 1E-3)
+        assert(model2a0.weights ~== model2b.weights absTol 1E-3)
+        assert(model2a0.intercept ~== model2b.intercept absTol 1E-3)
+
+        val trainer3a = (new LinearRegression).setFitIntercept(false)
+          .setElasticNetParam(0.38).setRegParam(0.21).setStandardization(true).setSolver(solver)
+        val trainer3b = (new LinearRegression).setFitIntercept(false).setWeightCol("weight")
+          .setElasticNetParam(0.38).setRegParam(0.21).setStandardization(true).setSolver(solver)
+        val model3a0 = trainer3a.fit(data)
+        val model3a1 = trainer3a.fit(weightedData)
+        val model3b = trainer3b.fit(weightedData)
+        assert(model3a0.weights !~= model3a1.weights absTol 1E-3)
+        assert(model3a0.weights ~== model3b.weights absTol 1E-3)
+
+        val trainer4a = (new LinearRegression).setFitIntercept(false)
+          .setElasticNetParam(0.38).setRegParam(0.21).setStandardization(false).setSolver(solver)
+        val trainer4b = (new LinearRegression).setFitIntercept(false).setWeightCol("weight")
+          .setElasticNetParam(0.38).setRegParam(0.21).setStandardization(false).setSolver(solver)
+        val model4a0 = trainer4a.fit(data)
+        val model4a1 = trainer4a.fit(weightedData)
+        val model4b = trainer4b.fit(weightedData)
+        assert(model4a0.weights !~= model4a1.weights absTol 1E-3)
+        assert(model4a0.weights ~== model4b.weights absTol 1E-3)
       }
-      val data2 = weightedSignedData ++ weightedNoiseData
-
-      (sqlContext.createDataFrame(sc.parallelize(data1, 4)),
-        sqlContext.createDataFrame(sc.parallelize(data2, 4)))
     }
+  }
+
+  test("LBFGS and normal optimizer should not have diffs") {
```

Comment:
Why this test is added? I thought this one is already covered. What missing is that comparing weighted sample version of `normal` and `l-bfgs` solver. See the session of  `test("linear regression with weighted samples"){`.